src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITYY = 10;
+    private static final double INCREMENT_INDEX = 1.5;
+    private T[] array = (T[]) new Object[DEFAULT_CAPACITYY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        resizeArray();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("You can't add element to index: "
+                    + index);
+        }
```

Comment:
extract to the separate method
```suggestion
        checkIndexForAdd(index);
```